### DISCUSSION TITLE
Are "Verification emails" on by default?

### DISCUSSION THREAD

**Discussion by charbs-io** [pocketbase/pocketbase#3140] [Q&A] [answered]
I cannot figure out why PocketBase is not sending verification emails when new `user` records are added.

I know my SMTP setup works because when I test it using the functionality on `/_/#/settings/mail` it works.

I have an `onRecordAfterCreateRequest` hook on my `users` collection - could that somehow be blocking the email from being sent or something?

**Comment by charbs-io** [comment]
Perhaps the simple answer is I have to explicitly call `pb.collection('users').requestVerification()` method explicitly. Which is totally fine - but I wanted to ask and ensure i haven't missed anything... 

Side question: Is there any way I can call `requestVerification` from the pb_hooks directory?

**Reply by ganigeorgiev** [reply, marked as answer]
You need to explicitly call `requestVerification()` to send the verification email.
The reason behind this was that users can be created on batches either by admin (or other "manager" type of users) and they may not want to trigger the email notification.

> Side question: Is there any way I can call requestVerification from the pb_hooks directory?

You can:
- call the related HTTP endpoint with the [`$http` client](https://pocketbase.io/docs/js-sending-http-requests/)
- initialize and submit the [`RecordVerificationRequestForm`](https://pocketbase.io/jsvm/classes/RecordVerificationRequestForm.html), something like:
    ```js
    const users = $app.dao().findCollectionByNameOrId("users")

    const form = new RecordVerificationRequestForm($app, users)

    form.email = "test@example.com"

    form.submit()
    ```

- or if you don't need validation with the next release I'll consider exposing the Go [`mails` functions](https://pkg.go.dev/github.com/pocketbase/pocketbase/mails) and you'll be able to do something like:
    ```js
    const user = $app.dao().findAuthRecordByEmail("users", "test@example.com")

    $mails.sendRecordVerification($app, user)
    ```
